### PR TITLE
Favorites frontend

### PR DESCRIPTION
**Pull request by owensimpson**
### Synopsis
Users can now click other users to favorite on the front end and it will store it in our database

### Changelog
- Created tRPC hook for "edit-favorites"
- Added func prop to sidebar and usercard components to handle the favoriting action
- added function to index.tsx to call the mutate endpoint and also update the favorites state

### Contributors
Me myself and I (Owen)

### CODE REVIEW

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -72,7 +73,9 @@ const Sidebar = (props: SideBarProps) => {
               previousMarkers: previousMarkers,
               clearMarkers: clearMarkers,
             }}
+            isFavorited={props.favs.map((fav) => fav.id).includes(otherUser.id)}
```
Comment (jun-w-huang) [inline src/components/Sidebar.tsx:76]: Wouldn't it be better to just create a const which is a Set of the favorited ids? That way we would only loop through props.favs to create the set, then access time is O(1). Currently, we are looping through favs for every recommended user, when we could just loop through once. Unless I am misunderstanding something

Comment (owensimpson) [reply src/components/Sidebar.tsx:76]: good point

File: src/pages/index.tsx [hunk 2]
```diff
@@ -61,6 +69,16 @@ const Home: NextPage<any> = () => {
       setMapState(newMap);
     }
   }, [user, geoJsonUsers]);
+
+  const addFavorite = (favoriteId: string, add: boolean) => {
```
Comment (jun-w-huang) [inline src/pages/index.tsx:73]: sorry to be nitpicky but can we rename this to handleFavorite just for readability?

**Review by jun-w-huang — DISMISSED** [review]
clean

**Pull request merged**